### PR TITLE
Cloud info: Add support for Oracle Cloud

### PR DESCRIPTION
**Pull request by uriariel**

### CODE REVIEW

File: granulate_utils/metadata/cloud.py [hunk 1]
```diff
@@ -151,6 +161,23 @@ def get_azure_metadata() -> Optional[AzureInstanceMetadata]:
...
+    response = send_request("http://169.254.169.254/opc/v2/instance/", headers={"Authorization": "Bearer Oracle"})
+    if response is None:
+        return None
+    instance = response.json()
+
+    return OracleCloudInstanceMetadata(
+        provider="oracle",
+        instance_type=instance["shape"],
+        region=instance["canonicalRegionName"],
+        memory=instance["shapeConfig"]["memoryInGBs"],
```
Comment (YoniKF) [inline granulate_utils/metadata/cloud.py:175]: Other cloud provider classes don't have this attribute (yet), let's see can get memory information on them as well.
Also, rename to include the memory units (`memory_mb` or `memory_gb`).

File: granulate_utils/metadata/cloud.py [hunk 2]
```diff
@@ -151,6 +161,23 @@ def get_azure_metadata() -> Optional[AzureInstanceMetadata]:
...
+        name=instance["hostname"],
```
Comment (YoniKF) [inline granulate_utils/metadata/cloud.py:176]: For GCP we access `name` and not `hostname`. Find out if it matters and make the behavior here consistent.

File: granulate_utils/metadata/cloud.py [hunk 3]
```diff
@@ -151,6 +161,23 @@ def get_azure_metadata() -> Optional[AzureInstanceMetadata]:
...
+        image=instance["image"],
```
Comment (YoniKF) [inline granulate_utils/metadata/cloud.py:177]: Other classes have an `image_id` field, why is this one `image`?

**Review by YoniKF — CHANGES_REQUESTED** [review]

**Comment by michelhe** [comment]
@YoniKF @uriariel Status here?

**Comment by YoniKF** [comment]
PR is getting stale, closing.

**Pull request closed without merging**